qiskit/circuit/library/pauli_evolution.py

```diff
@@ -356,3 +356,88 @@ def _get_default_label(operator):
     if isinstance(operator, list):
         return "exp(-it [" + ", ".join(_operator_label(op) for op in operator) + "])"
     return f"exp(-it {_operator_label(operator)})"
+
+
+def _merge_two_pauli_evolutions(
+    gate1: PauliEvolutionGate, gate2: PauliEvolutionGate
+) -> PauliEvolutionGate | None:
+    """
+    Attempts to merge two PauliEvolutionGates can be merged.
+
+    Returns:
+
+    * None if the arguments are not of type PauliEvolutionGate or cannot be merged,
+    * Combined PauliEvolutionGate otherwise.
+
+    This function is internal (used from within Rust code) and not a part of public API.
+    """
+    if not isinstance(gate1, PauliEvolutionGate) or not isinstance(gate2, PauliEvolutionGate):
+        return None
+
+    if isinstance(gate1.operator, SparseObservable) and isinstance(
+        gate2.operator, SparseObservable
+    ):
+        # When both operators are SparseObservables, we can compare their canonical representatives.
+        can_merge = gate1.operator.simplify() == gate2.operator.simplify()
+    elif isinstance(gate1.operator, SparsePauliOp) and isinstance(gate2.operator, SparsePauliOp):
+        # SparsePauliOp already has a function that compares canonical representatives.
+        can_merge = gate1.operator.equiv(gate2.operator)
+    else:
+        can_merge = gate1.operator == gate2.operator
+
+    if can_merge:
+        return PauliEvolutionGate(gate1.operator, gate1.time + gate2.time)
+
+    return None
+
+
+# pylint: disable=too-many-return-statements
+def pauli_rotation_trace_and_dim(gate: PauliEvolutionGate) -> tuple[complex, int] | None:
```

Comment:
Do you want to add a leading underscore?

Reply:
Done in [c2306c3](https://github.com/Qiskit/qiskit/pull/15047/commits/c2306c3a72ee33cd32ed964d0151892f289e2f0a).